src/main/java/org/ice4j/ice/Component.java

```diff
@@ -902,6 +922,12 @@ public RemoteCandidate findRemoteCandidate(TransportAddress remoteAddress)
      */
     protected void setSelectedPair(CandidatePair pair)
     {
+        if (keepAliveStrategy == KeepAliveStrategy.SELECTED_ONLY)
```

Comment:
or, if we don't want to let it be changed, maybe we should just pass the keep-alive strategy in the ctor rather than having a setter?